libassuan: 2.5.7 -> 3.0.2

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for libassuan is: IPC library used by GnuPG and related software

meta.homepage for libassuan is: https://gnupg.org/software/libassuan/

meta.changelog for libassuan is: https://dev.gnupg.org/source/libassuan/browse/master/NEWS;libassuan-3.0.1


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 3.0.1 with grep in /nix/store/0mm0rniklbabma3rm16xd018jsbka8g9-libassuan-3.0.1
- found 3.0.1 in filename of file in /nix/store/0mm0rniklbabma3rm16xd018jsbka8g9-libassuan-3.0.1

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
25961 total rebuild path(s)

25960 package rebuild(s)

First fifty rebuilds by attrpath

CuboCore.coreaction
CuboCore.corearchiver
CuboCore.corefm
CuboCore.coregarage
CuboCore.corehunt
CuboCore.coreimage
CuboCore.coreinfo
CuboCore.corekeyboard
CuboCore.corepad
CuboCore.corepaint
CuboCore.corepdf
CuboCore.corepins
CuboCore.corerenamer
CuboCore.coreshot
CuboCore.corestats
CuboCore.corestuff
CuboCore.coreterminal
CuboCore.coretime
CuboCore.coretoppings
CuboCore.coreuniverse
CuboCore.libcprime
CuboCore.libcsys
DisnixWebService
LPCNet
MMA
OVMF
OVMF-cloud-hypervisor
OVMFFull
R
SDL2
SDL2_Pango
SDL2_gfx
SDL2_image
SDL2_image_2_0
SDL2_image_2_6
SDL2_mixer
SDL2_mixer_2_0
SDL2_net
SDL2_sound
SDL2_ttf
SDL_audiolib
SDL_compat
SDL_gpu
SDL_mixer
_010editor
_1oom
_1password-gui
_1password-gui-beta
_20kly
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/0mm0rniklbabma3rm16xd018jsbka8g9-libassuan-3.0.1 \
  --option binary-caches 'https://cache.nixos.org/ https://nix-community.cachix.org/' \
  --option trusted-public-keys '
  nix-community.cachix.org-1:mB9FSh9qf2dCimDSUo8Zy7bkq5CX+/rkCWyvRCYg3Fs=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A libassuan https://github.com/r-ryantm/nixpkgs/archive/53b036ad4e29f409ef33bc5a184322ca1e90514b.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/53b036ad4e29f409ef33bc5a184322ca1e90514b#libassuan
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/0mm0rniklbabma3rm16xd018jsbka8g9-libassuan-3.0.1
ls -la /nix/store/0mm0rniklbabma3rm16xd018jsbka8g9-libassuan-3.0.1/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings



> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda commented on pkgs/by-name/li/libassuan/package.nix:
```diff
@@ -11,11 +11,11 @@
   src = fetchurl {
     url = "mirror://gnupg/${pname}/${pname}-${version}.tar.bz2";
-    sha256 = "sha256-AQMIH/wng4ouUEeRU8oQXoc9PWXYqVkygunJTH5q+3Y=";
+    sha256 = "sha256-yPD0LmED3qSxpqSDy1VmVOlzAsdGUwj1g2N3j5XxlLE=";
```
```suggestion
    hash = "sha256-yPD0LmED3qSxpqSDy1VmVOlzAsdGUwj1g2N3j5XxlLE=";
```

dotlambda commented on pkgs/by-name/li/libassuan/package.nix:
```diff
@@ -11,11 +11,11 @@
+  version = "3.0.1";
 
   src = fetchurl {
     url = "mirror://gnupg/${pname}/${pname}-${version}.tar.bz2";
```
```suggestion
    url = "mirror://gnupg/libassuan/libassuan-${version}.tar.bz2";
```
#277994

dotlambda committed:
gnupg-pkcs11-scd: 0.10.0 -> 0.11.0

Diff: https://github.com/alonbl/gnupg-pkcs11-scd/compare/gnupg-pkcs11-scd-0.10.0...gnupg-pkcs11-scd-0.11.0

Changelog: https://github.com/alonbl/gnupg-pkcs11-scd/blob/gnupg-pkcs11-scd-0.11.0/ChangeLog

dotlambda committed:
gpa: 0.10.0 -> 0.11.0

Changelog: https://dev.gnupg.org/source/gpa/browse/master/NEWS%3Bgpa-0.11.0?view=raw

dotlambda commented:
`pinentry_mac` fails to build with
```
configure: WARNING: fltk-config is not found
configure:
***
*** You need libassuan to build this program.
*** This library is for example available at
***   ftp://ftp.gnupg.org/gcrypt/libassuan/
*** (at least version 2.1.0 (API 2) is required).
***
configure: error: 
***
*** Required libraries not found. Please consult the above messages
*** and install them before running configure again.
***
```
cc @reckenrode @midchildan

reckenrode committed:
pinentry_mac: fix build with libassuan 3.0

The autoconf macros shipped with the source need updating to correctly
detect libassuan 3.0. Once they are updated, pinentry_mac builds again.

reckenrode commented:
pinentry_mac needs an updated `libassuan.m4` from libassuan to properly detect libassuan 3 as supporting the v2 API. I pushed a commit to the branch with the fix.

emilazy commented:
Is this ready to go?

r-ryantm committed:
libassuan: 2.5.7 -> 3.0.2

r-ryantm commented:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for libassuan is: IPC library used by GnuPG and related software

meta.homepage for libassuan is: https://gnupg.org/software/libassuan/

meta.changelog for libassuan is: https://dev.gnupg.org/source/libassuan/browse/master/NEWS;libassuan-3.0.2


###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes





###### Impact

<b>Checks done</b>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 3.0.2 with grep in /nix/store/p663zh44w54ir1h6qz7xl72zkgmfgif0-libassuan-3.0.2
- found 3.0.2 in filename of file in /nix/store/p663zh44w54ir1h6qz7xl72zkgmfgif0-libassuan-3.0.2

---

<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
27197 total rebuild path(s)

27196 package rebuild(s)

First fifty rebuilds by attrpath

CuboCore.coreaction
CuboCore.corearchiver
CuboCore.corefm
CuboCore.coregarage
CuboCore.corehunt
CuboCore.coreimage
CuboCore.coreinfo
CuboCore.corekeyboard
CuboCore.corepad
CuboCore.corepaint
CuboCore.corepdf
CuboCore.corepins
CuboCore.corerenamer
CuboCore.coreshot
CuboCore.corestats
CuboCore.corestuff
CuboCore.coreterminal
CuboCore.coretime
CuboCore.coretoppings
CuboCore.coreuniverse
CuboCore.libcprime
CuboCore.libcsys
DisnixWebService
LPCNet
MMA
OVMF
OVMF-cloud-hypervisor
OVMFFull
R
SDL2
SDL2_Pango
SDL2_gfx
SDL2_image
SDL2_image_2_0
SDL2_image_2_6
SDL2_mixer
SDL2_mixer_2_0
SDL2_net
SDL2_sound
SDL2_ttf
SDL_audiolib
SDL_compat
SDL_gpu
SDL_mixer
_010editor
_1oom
_1password-gui
_1password-gui-beta
_20kly
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from the cache**:
```
nix-store -r /nix/store/p663zh44w54ir1h6qz7xl72zkgmfgif0-libassuan-3.0.2 \
  --option binary-caches 'https://cache.nixos.org/ https://nixpkgs-update-cache.nix-community.org/' \
  --option trusted-public-keys '
  nixpkgs-update-cache.nix-community.org-1:U8d6wiQecHUPJFSqHN9GSSmNkmdiFW7GW7WNAnHW0SM=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(The nixpkgs-update cache is only trusted for this store-path realization.)
For the cached download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A libassuan https://github.com/r-ryantm/nixpkgs/archive/2092b1f45c650e07c4ef96fa38fbd8f2cae53c56.tar.gz
```
Or:
```
nix build github:r-ryantm/nixpkgs/2092b1f45c650e07c4ef96fa38fbd8f2cae53c56#libassuan
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/p663zh44w54ir1h6qz7xl72zkgmfgif0-libassuan-3.0.2
ls -la /nix/store/p663zh44w54ir1h6qz7xl72zkgmfgif0-libassuan-3.0.2/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings



> [!TIP]
> As a maintainer, if your package is located under `pkgs/by-name/*`, you can comment **`@NixOS/nixpkgs-merge-bot merge`** to automatically merge this update using the [`nixpkgs-merge-bot`](https://github.com/NixOS/nixpkgs-merge-bot).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

reckenrode commented:
I restored the commits fixing breakage. Is this ready to go?

dotlambda committed:
libassuan: modernize

dotlambda committed:
gpa: fix build

dotlambda reviewed (approved):
Every package mentioning `libassuan` in its expression builds on x86_64-linux.
